### PR TITLE
JBPM-9659: Upgrade deployments to WildFly 23

### PR DESCRIPTION
**Pull request by jesuino**
**JIRA**: 

[JBPM-9659](https://issues.redhat.com/browse/JBPM-9659)

**Related PRs*: 

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1693
https://github.com/kiegroup/kie-soup/pull/216
https://github.com/kiegroup/droolsjbpm-knowledge/pull/543
https://github.com/kiegroup/drools/pull/3757
https://github.com/kiegroup/optaplanner/pull/1463
https://github.com/kiegroup/lienzo-tests/pull/124
https://github.com/kiegroup/appformer/pull/1163
https://github.com/kiegroup/kie-uberfire-extensions/pull/122
https://github.com/kiegroup/jbpm/pull/1974
https://github.com/kiegroup/kie-jpmml-integration/pull/62
https://github.com/kiegroup/droolsjbpm-integration/pull/2539
https://github.com/kiegroup/kie-wb-playground/pull/115
https://github.com/kiegroup/kie-wb-common/pull/3660
https://github.com/kiegroup/drools-wb/pull/1506
https://github.com/kiegroup/jbpm-designer/pull/913
https://github.com/kiegroup/jbpm-work-items/pull/212
https://github.com/kiegroup/jbpm-wb/pull/1506
https://github.com/kiegroup/optaplanner-wb/pull/399
https://github.com/kiegroup/kie-wb-distributions/pull/1115
https://github.com/kiegroup/openshift-drools-hacep/pull/116
https://github.com/kiegroup/process-migration-service/pull/5
https://github.com/kiegroup/optaweb-employee-rostering/pull/662
https://github.com/kiegroup/optaweb-vehicle-routing/pull/535

### CODE REVIEW

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute cdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
Jenkins run fdb

**Comment by tiagobento** [comment]
Jenkins run fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins run jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute fdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute fdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb

**Comment by jesuino** [comment]
jenkins execute cdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute jdk11-cdb

**Comment by jesuino** [comment]
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute fdb
jenkins execute jdk11-fdb

**Review by mbiarnes — APPROVED** [review]
+1

**Comment by jesuino** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Comment by jesuino** [comment]
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by tiagobento** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

**Comment by jesuino** [comment]
jenkins execute cdb
jenkins execute fdb
jenkins execute jdk11-cdb
jenkins execute jdk11-fdb

File: pom.xml [hunk 1]
```diff
@@ -314,32 +309,31 @@
...
-    <version.jakarta.json.bind>1.0.2</version.jakarta.json.bind>
     <version.org.antlr4>4.8</version.org.antlr4>
-    <version.org.apache.cxf>3.2.14</version.org.apache.cxf>
+    <!-- CXF's version is aligned with Jetty -->
+    <version.org.apache.cxf>3.3.9</version.org.apache.cxf>
     <version.org.apache.camel>2.24.0</version.org.apache.camel>
-    <version.org.codehaus.cargo>1.7.11</version.org.codehaus.cargo>
-    <version.org.freemarker>2.3.30</version.org.freemarker>
+    <version.org.codehaus.cargo>1.9.3</version.org.codehaus.cargo>
+    <version.org.freemarker>2.3.28</version.org.freemarker>
```
Comment (mareknovotny) [inline pom.xml:317]: why is here the freemarker downgrade? the previous version is a CVE fix was due https://app.snyk.io/vuln/SNYK-JAVA-ORGFREEMARKER-1076795

File: pom.xml [hunk 2]
```diff
@@ -435,7 +428,7 @@
     <version.org.kie.workbench.app>${version.org.kie}</version.org.kie.workbench.app>
 
     <version.thorntail>2.7.0.Final</version.thorntail>
-    <version.net.byte-buddy>1.10.3</version.net.byte-buddy>
+    <version.net.byte-buddy>1.9.11</version.net.byte-buddy>
```
Comment (mareknovotny) [inline pom.xml:431]: not sure this downgrade is ok, see why it was upgraded before as Mockito tests were failing with this older version https://github.com/kiegroup/droolsjbpm-build-bootstrap/commit/0ebdb10c3ddb06f371e91b6edc04d8aae101a900

Comment (jesuino) [reply pom.xml]: Hello @mareknovotny  I reverted the downgrade in 1dd4fcf7849e8eeb3b927b89384286e546c77c74 - Thanks!

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb
jenkins execute fdb

**Comment by tiagobento** [comment]
jenkins execute cdb

**Review by tomasdavidorg — APPROVED** [review]

**Comment by tomasdavidorg** [comment]
QE tests passed with the WARs from FDB. Approving this PR.

**Comment by mareknovotny** [comment]
I guess failed tests are random failures or instability in them ?


- org.kie.workbench.common.services.backend.compiler.rest.client.MavenRestClientTest.org.kie.workbench.common.services.backend.compiler.rest.client.MavenRestClientTest
- org.kie.workbench.common.services.backend.compiler.offprocess.CompilerChronicleCoordinatorTest.offProcessTwoBuildTest
- org.kie.workbench.common.services.backend.compiler.offprocess.CompilerChronicleCoordinatorTest.offProcessOneBuildTest
- org.kie.workbench.common.services.backend.compiler.offprocess.service.CompilerOffProcessServiceTest.offProcessServiceCompileAsyncTest
- org.kie.workbench.common.services.datamodel.backend.server.ModuleDataModelConcurrencyTest.testConcurrentResourceUpdates

**Review by mareknovotny — APPROVED** [review]

**Comment by tiagobento** [comment]
jenkins execute fdb

**Comment by tiagobento** [comment]
@mareknovotny They look like random failures. We haven't seen failures on those so far.

**Comment by tiagobento** [comment]
@jesuino @mareknovotny Green FDB!!

**Pull request merged**